File: app/src/main/java/com/ddu/ui/fragment/study/imitate/UIFragment.java
package com.ddu.ui.fragment.study.imitate;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.AdapterView;
import android.widget.FrameLayout;
import android.widget.ScrollView;

import com.ddu.icore.ui.adapter.common.DefaultListViewAdapter;
import com.ddu.icore.ui.adapter.common.ViewHolder;
import com.ddu.icore.ui.fragment.DefaultFragment;
import com.ddu.icore.util.FragmentUtils;
import com.ddu.R;
import com.ddu.ui.fragment.study.ui.ShapeFragment;
import com.ddu.ui.view.ListViewForScrollView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by yzbzz on 16/4/8.
 */
public class UIFragment extends DefaultFragment implements AdapterView.OnItemClickListener {


    @Nullable
    @BindView(R.id.lv_ui_navigation)
    ListViewForScrollView lvUiNavigation;
    @Nullable
    @BindView(R.id.ls_ui_navigation)
    ScrollView lsUiNavigation;
    @Nullable
    @BindView(R.id.fl_ui_detail)
    FrameLayout flUiDetail;

    private RecyclerView.LayoutManager mLayoutManager;

    private List<String> mList;

    private Unbinder unbinder;

    @Override
    public void initData(Bundle savedInstanceState) {
        mList = new ArrayList<>();
        for (int i = 0; i < 20; i++) {
            mList.add(i + "");
        }
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_ui;
    }

    @Override
    public void initView() {
        unbinder = ButterKnife.bind(this, mView);

        lvUiNavigation.setAdapter(new DefaultListViewAdapter<String>(mContext,mList) {
            @Override
            public void bindView(int position, String s, @NonNull ViewHolder holder) {
                holder.setText(R.id.tv_navi_name, s);
            }

            @Override
            public int getLayoutId() {
                return R.layout.fragment_ui_navi_item;
            }
        });

        lvUiNavigation.setOnItemClickListener(this);
        setDefaultTitle("UI");
    }

    public void initTitle() {
        setDefaultTitle("UI");
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }

    @Override
    public void onItemClick(AdapterView<?> parent, @NonNull View view, int position, long id) {
        lsUiNavigation.smoothScrollTo(0, position * view.getHeight());
        ShapeFragment shapeFragment = ShapeFragment.newInstance("");
        FragmentUtils.replaceFragment(baseActivity.getSupportFragmentManager(), shapeFragment, R.id.fl_ui_detail);
    }
}

File: app/src/main/java/com/ddu/ui/fragment/study/imitate/UIFragment.kt
package com.ddu.ui.fragment.study.imitate

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.AdapterView
import com.ddu.R
import com.ddu.icore.ui.adapter.common.DefaultListViewAdapter
import com.ddu.icore.ui.adapter.common.ViewHolder
import com.ddu.icore.ui.fragment.DefaultFragment
import com.ddu.icore.util.FragmentUtils
import com.ddu.ui.fragment.study.ui.DrawViewFragment
import com.ddu.ui.fragment.study.ui.ShapeFragment
import com.iannotation.Element
import kotlinx.android.synthetic.main.fragment_ui.*

/**
 * Created by yzbzz on 16/4/8.
 */
@Element("UI")
class UIFragment : DefaultFragment(), AdapterView.OnItemClickListener {

    private val mLayoutManager: RecyclerView.LayoutManager? = null

    private var mList: MutableList<Fragment> = mutableListOf()

    init {
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(ShapeFragment.newInstance(""))
        mList.add(DrawViewFragment::class.java.newInstance())
        mList.add(DrawViewFragment::class.java.newInstance())
    }


    override fun initData(savedInstanceState: Bundle?) {

    }

    override fun getLayoutId(): Int {
        return R.layout.fragment_ui
    }

    override fun initView() {
        lv_ui_navigation.adapter = object : DefaultListViewAdapter<Fragment>(mContext, mList) {
            override fun bindView(position: Int, s: Fragment, holder: ViewHolder) {
                holder.setText(R.id.tv_navi_name, s::class.java.simpleName)
            }

            override fun getLayoutId(): Int {
                return R.layout.fragment_ui_navi_item
            }
        }

        lv_ui_navigation.onItemClickListener = this
        setDefaultTitle("UI")
    }


    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        ls_ui_navigation!!.smoothScrollTo(0, position * view.height)
        val fragment = parent.adapter.getItem(position) as? Fragment
        fragment?.apply {
            FragmentUtils.replaceFragment(baseActivity.supportFragmentManager, fragment, R.id.fl_ui_detail)
        }
    }
}
